test/testPartyAttachments.js

```diff
@@ -0,0 +1,1025 @@
+      body,
+      streaming: true, // ask for streaming response
+    });
+
+    const response = {
+      ...responseStreaming,
+      body: (await getStreamAsBuffer(responseStreaming.body)).toString(),
+    }
+
+    assert.equal(response.status, 200);
+
+    await checkUploadedFiles(httpClient, filesToPut);
+  } else {
+    assert.fail("[copyFilesAndCheck]: filesToPut is empty or not defined.");
+  }
+}
+
+exports = module.exports = function (httpClient, type) {
+  describe(type, function () {
+    describe("PUT (customRouteForUpload)", function () {
+      // checks customRouteForUpload, customRouteForDownload and customRouteForDelete
+      it("should allow adding of profile picture as attachment", async () => {
+        const body = {
+          type: "person",
+          name: "test user",
+          status: "active",
+        };
+        const resourceKey = uuid.v4();
+        const resourceHref = type + "/" + resourceKey;
+        const attachmentKey = uuid.v4();
+
+        const responsePut = await httpClient.put({ path: resourceHref, body});
+        assert.equal(responsePut.status, 201);
+
+        // Add attachment
+        debug("PUTting the profile image as attachment");
+        const filesToPut = [
+          {
+            remotefileName: "profile.png",
+            localFilename: "test/images/orange-boy-icon.png",
+            attachmentKey,
+            resourceHref,
+          },
+        ];
+        await uploadFilesAndCheck(httpClient, filesToPut);
+
+        // Overwrite attachment
+        const filesToPut2 = [
+          {
+            remotefileName: "profile.png",
+            localFilename: "test/images/little-boy-white.png",
+            attachmentKey,
+            resourceHref,
+          },
+        ];
+        await uploadFilesAndCheck(httpClient, filesToPut2);
+
+        const attachmentUrl =
+          type + "/" + resourceKey + "/attachments/" + attachmentKey;
+        const attachmentDownloadUrl =
+          type + "/" + resourceKey + "/attachments/profile.png";
+
+        // Next : try to delete the resource.
+        const response6 = await httpClient.delete({ path: attachmentUrl });
+        assert.equal(response6.status, 204);
+        // Now check that is is gone..
+        const response7 = await httpClient.get({ path: attachmentDownloadUrl });
+        assert.equal(response7.status, 404);
+      });
+
+      it("should be idempotent", async function () {
+        const body = {
+          type: "person",
+          name: "test user",
+          status: "active",
+        };
+        const resourceKey = uuid.v4();
+        const resourceHref = type + "/" + resourceKey;
+        const attachmentKey = uuid.v4();
+        const attachmentUrl =
+          type + "/" + resourceKey + "/attachments/profile.png";
+
+        debug("Generated UUID=" + resourceKey);
+        const response = await httpClient.put({ path: resourceHref, body });
+        assert.equal(response.status, 201);
```

Comment:
This is the same resource as in the previous test, can't this data setup be shared in a `before` or `beforeEach` hook? And then cleaned up in an `after` or `afterEach` hook?